addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusData.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.tivo.internal.service;
+
+import java.util.Date;
+
+/**
+ * {@link TivoStatusData} class stores the data from the last status query from the TiVo and any other errors / status
+ * codes.
+ *
+ * @param cmdOk boolean true = last command executed correctly, false = last command failed with error message
+ * @param channelNum int = channel number, -1 indicates no channel received. Valid channel range 1-9999.
+ * @param msg string status message from the TiVo socket
+ * @param pubToUI boolean true = this status needs to be published to the UI / Thing, false = do not publish (or it
+ *            already has been)
+ * @param chScan boolean true = channel scan is in progress, suspend polling actions
+ * @param connOK int -1= test not run/default, 0 = offline, 1 = both connection tests failed, 2 = TiVo is in standby
+ *            (write test only passed), 3 = Online, both connection tests passed
+ *
+ */
+
+public class TivoStatusData {
+    private boolean cmdOk = false;
+    private Date time = new Date();
+    private int channelNum = -1;
+    private String msg = "NO STATUS QUERIED YET";
+    private boolean pubToUI = true;
+    private boolean chScan = false;
+    private int connOK = -1;
```

Comment:
`connOK` should be `connectionStatus` and return the enum value which I proposed for the `TivoStatusProvider#connTestCmds`.